File: f#_pi_dir/pi.fs
open System

let y(x) = 1.0/(1.0+(x*x))
let calcpi(n) = (4.0/n) * Seq.sum (Seq.map y ((seq { for i in 1.0..n -> i/n})))

[<EntryPoint>]
let Main args =
  let m = if args.Length > 0 then int args.[0] else 100000000
  let n = double m
  printfn "Calculating PI using:"
  printfn "  %i slices" m 
  let start = DateTime.Now
  let p = calcpi(n)
  printfn "Obtained value of PI: %f" p
  let stop = DateTime.Now
  let difference = stop.Subtract(start)
  let time = difference.TotalSeconds
  printfn "Total time taken: %f seconds" time
  0

File: python3_numba_pi_dir/pi.py
# Version of the Python Pi example written to use numba.

# Dr Owain Kenway

import time
import sys
import numba

# Can't use sum as it makes numba puke.
@numba.jit
def calcpi(n):
	step = 1.0/n
	t = 0.0
	for i in range(n):
		x = (i + 0.5) * step
		t += 4.0/(1.0 + (x * x))
	return t * step	
	

num_steps = 10000000

if len(sys.argv) > 1:
	num_steps = int(sys.argv[1])

print(f"Calculating PI with:\n  {num_steps} slices")

start = time.time()

p = calcpi(num_steps)

stop = time.time()

print(f"Obtained value of PI: {p}")
print(f"Time taken: {stop - start} seconds")
